src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -499,13 +528,14 @@ private void ParseChunkedPrefix(ReadableBuffer buffer, out ReadCursor consumed,
                 var chunkSize = CalculateChunkSize(ch1, 0);
                 ch1 = ch2;
 
-                do
+                while (reader.ConsumedBytes < 10)
                 {
                     if (ch1 == ';')
                     {
                         consumed = reader.Cursor;
                         examined = reader.Cursor;
 
+                        AddAndCheckConsumedBytes(reader.ConsumedBytes);
```

Comment:
Counting the framing is going to cause strange inconsistencies between Chunked, Content-Length, and HttpSysServer. @shirhatti I assume IIS does not count the chunked framing because it doesn't even see it, Http.Sys removes it, right?